src/Tests/CodeStandards/NamingConventions.doc.cs

```diff
@@ -12,59 +12,114 @@ public class NamingConventions
 		/**
 		* Abstract class names should end with a `Base` suffix
 		*/
-		//[U]
+		[U]
 		public void AbstractClassNamesEndWithBase()
 		{
+		    var exceptions = new[]
+		    {
+		        typeof(DateMath)
+		    };
+
 			var abstractClasses = typeof(IRequest).Assembly().GetTypes()
-				.Where(t => t.IsClass() && t.IsAbstract() && !t.IsSealed())
-				.Select(t => t.Name.Split('`')[0])
+				.Where(t => t.IsClass() && t.IsAbstract() && !t.IsSealed() && !exceptions.Contains(t))
+                .Where(t => !t.Name.Split('`')[0].EndsWith("Base"))
+                .Select(t => t.Name.Split('`')[0])
 				.ToList();
 
-			foreach (var abstractClass in abstractClasses)
-				abstractClass.Should().EndWith("Base");
+		    abstractClasses.Should().BeEmpty();
 		}
 
 		/**
 		* Request class names should end with "Request"
 		*/
-		//[U]
+		[U]
 		public void RequestClassNamesEndWithRequest()
 		{
 			var types = typeof(IRequest).Assembly().GetTypes();
 			var requests = types
-				.Where(t => typeof(IRequest).IsAssignableFrom(t))
-				.Select(t => t.Name.Split('`')[0])
+				.Where(t => typeof(IRequest).IsAssignableFrom(t) && !t.IsAbstract())
+                .Where(t => !typeof(IDescriptor).IsAssignableFrom(t))
+				.Where(t => !t.Name.Split('`')[0].EndsWith("Request"))
+                .Select(t => t.Name.Split('`')[0])
 				.ToList();
-			foreach (var request in requests)
-				request.Should().EndWith("Request");
+
+		    requests.Should().BeEmpty();
 		}
 
 		/**
 		* Response class names should end with "Response"
 		**/
-		//[U]
+		[U]
 		public void ResponseClassNamesEndWithResponse()
 		{
 			var types = typeof(IRequest).Assembly().GetTypes();
 			var responses = types
-				.Where(t => typeof(IResponse).IsAssignableFrom(t))
-				.Select(t => t.Name.Split('`')[0])
+				.Where(t => typeof(IResponse).IsAssignableFrom(t) && !t.IsAbstract())
+                .Where(t => !t.Name.Split('`')[0].EndsWith("Response"))
+                .Select(t => t.Name.Split('`')[0])
 				.ToList();
-			foreach (var response in responses)
-				response.Should().EndWith("Response");
+
+		    responses.Should().BeEmpty();
 		}
 
 		/**
 		* Request and Response class names should be one to one in *most* cases.
 		* e.g. ValidateRequest => ValidateResponse, and not ValidateQueryRequest => ValidateResponse
 		*/
-		//[U]
+		[U]
 		public void ParityBetweenRequestsAndResponses()
 		{
-			var types = typeof(IRequest).Assembly().GetTypes();
+            // Add any exceptions to the rule here
```

Comment:
I feel the exceptions should be based on response not request for a lot of these but thats a total nitpick :)